core/core.go

```diff
@@ -281,12 +281,21 @@ func Password(jsonIn []byte) ([]byte, error) {
 // Encrypt processes an encrypt request.
 func Encrypt(jsonIn []byte) ([]byte, error) {
 	var s EncryptRequest
-	if err := json.Unmarshal(jsonIn, &s); err != nil {
+
+	err := json.Unmarshal(jsonIn, &s)
+	if err != nil {
 		return jsonStatusError(err)
 	}
 
-	if err := validateUser(s.Name, s.Password, false); err != nil {
-		log.Printf("failed to validate user %s in request to encrypt: %v", s.Name, err)
+	defer func() {
+		if err != nil {
```

Comment:
I played with a little bit with defer closure to fully understand this trick. Thanks @kisom.